tests/functional/test_awsclient.py

```diff
@@ -475,6 +491,75 @@ def test_can_provide_tags(self, stubbed_session):
             handler='app.app') == 'arn:12345:name'
         stubbed_session.verify_stubs()
 
+    def test_raises_large_deployment_error_for_connection_error(
+            self, mock_lambda_client):
+        mock_session = mock.Mock(botocore.session.Session)
```

Comment:
Why are we doing this?? There's an existing way to test this, which is using the StubbedSession.

If the necessary abstractions aren't in place in StubbedSession, I think it's more appropriate it to add them rather than work around them bypassing the StubbedSession and dropping down to botocore sessions directly.